Python/sysmodule.c

```diff
@@ -1643,6 +1643,7 @@ static PyObject *
 _sys_getwindowsversion_from_kernel32(void)
 {
 #ifndef MS_WINDOWS_DESKTOP
+    PyErr_SetFromWindowsErr(0);
```

Comment:
I'm not sure it is correct. If you pass 0 then `PyErr_SetExcFromWindowsErrWithFilename` will try to get error from `GetLastError`, but in this place it may have different value.

Reply:
Yeah, we should probably just use an explicit error here:

```suggestion
    PyErr_SetString(PyExc_OSError, "cannot read version info on this platform");
```

Reply:
good catch :+1: